modules/fun.js

```diff
@@ -0,0 +1,450 @@
+    const response = await axios({ url, method: `get` }).catch((/** @type {axios.AxiosError} */ e) => {
+      int.editReply(`Could not generate lyrics for ${name}.\nPerhaps you can get it yourself from https://thenamegame-generator.com.`);
+      throw new Error(`namegame command error:${e.status}`);
+    });
+    const data = response.data;
+    if (data) {
+      const profanityFilter = require(`profanity-matcher`);
+      const pf = new profanityFilter();
+      const lyricsUntrimmedEnd = data.substring(data.indexOf(`<blockquote>`) + 12);
+      const lyricsTrimmedWithHtml = lyricsUntrimmedEnd.substring(0, lyricsUntrimmedEnd.indexOf(`</blockquote>`));
+      const results = lyricsTrimmedWithHtml.replace(/<br>/g, `\n`).replace(/<br \/>/g, `\n`);
+      const pfresults = pf.scan(results.toLowerCase().replace(/[-\n]/g, ` `).replace(/\s\s+/g, ` `));
+      const ispf = (pfresults.length > 0 && pfresults[0]) || (pfresults.length > 1);
+      if (!ispf && (name.length <= 230) && (results.length + name.length <= 5750)) {
+        const embed = u.embed().setTitle(`🎶 **The Name Game! ${name}! 🎵`).setDescription(results);
+        int.editReply({ embeds:[embed] });
+      } else {
+        int.editReply(`😬`);
+      }
+    } else {
+      int.editReply(`❌`);
+    }
+  } catch (error) { u.errorHandler(error, int); }
+}
+
+/**
+ * function slashFunChoose
+ * @param {Discord.ChatInputCommandInteraction} int a /fun choose interaction
+ */
+async function slashFunChoose(int) {
+  const optionsArg = int.options.getString(`options`);
+  if (optionsArg && optionsArg.includes("|")) {
+    const options = optionsArg.split("|");
+    const prefixes = ["I choose", "I pick", "I decided"];
+    return int.editReply(`${u.rand(prefixes)} **${u.rand(options)}**`);
+  } else {
+    return int.editReply("you need to give me two or more choices! `a | b`");
+  }
+}
+const Module = new Augur.Module()
+.addInteraction({
+  name: `fun`,
+  id: u.sf.commands.slashFun,
+  process: async (int) => {
+    const subcommand = int.options.getSubcommand(true);
+    await int.deferReply(); // { ephemeral: true });
+    switch (subcommand) {
+      case `roll`: return slashFunRoll(int);
+      case `8ball`: return slashFun8ball(int);
+      case `repost`: return slashFunRepost(int);
+      case `mines`: return slashFunMinesweeper(int);
+      case `acronym`: return slashFunAcronym(int);
+      case `hbs`: return slashFunHBS(int);
+      case `color`: return slashFunColor(int);
+      case `buttermelon`: return slashFunButtermelon(int);
+      case `quote`: return slashFunQuote(int);
+      case `namegame`: return slashFunNamegame(int);
+      case `choose`: return slashFunChoose(int);
+      default:
+        u.errorLog.send({ embeds: [ u.embed().setDescription(`Error, command ${int} isn't associated with anything in fun.js`)] });
+        return int.editReply(`Thats an error, this command isn't registered properly. I've let my devs know.`);
```

Comment:
We have a built in error handler. Just do `throw new Error("Unhandled Subcommand")` and it'll take care of the rest